osquery/tables/applications/chrome/chrome_extension_content_scripts.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/logger/logger.h>
+#include <osquery/tables/applications/chrome/utils.h>
+
+namespace osquery {
+
+namespace tables {
+
+QueryData genChromeExtensionContentScripts(QueryContext& context) {
+  auto profile_list = getChromeProfiles(context);
+
+  QueryData results;
+
+  for (const auto& profile : profile_list) {
+    Row row = {};
+    row["browser_type"] = SQL_TEXT(getChromeBrowserName(profile.type));
+    row["uid"] = BIGINT(profile.uid);
+    row["profile"] = SQL_TEXT(profile.name);
+    row["profile_path"] = SQL_TEXT(profile.path);
+
+    for (const auto& extension : profile.extension_list) {
+      std::string version = {};
+
+      auto version_it = extension.properties.find("version");
+      if (version_it == extension.properties.end()) {
+        LOG(ERROR)
+            << "The version property for the following extension is missing: "
+            << extension.path;
+
+      } else {
+        version = version_it->second;
+      }
+
+      row["version"] = SQL_TEXT(version);
```

Comment:
Should this be in the if block?

Reply:
The intent here was to always initialize the column; in case the version field is not present in the properties, it will initialize the version column to an empty string

Reply:
So this has the effect of setting it to an empty string, instead of leaving it null? I'm not sure I have a strong feeling either way. Initially, null seems more correct.

I guess I'd have ended up with:
```
      if (version_it == extension.properties.end()) {
        LOG(ERROR)
            << "The version property for the following extension is missing: "
            << extension.path;
        row["version"] = ""
      } else {
        row["version"] = SQL_TEXT(version_it->second);
      }
```

Reply:
In the past we had some crashes due to code assuming that columns were present when in fact they were being left missing by the implementation; this gets harder when the implementation differs based on platform. Since then, I think we have all been trying to make sure that table code always populate all the columns listed in the schema

Reply:
I remember issues based on whether or not a column is defined, platform specific columns end up being very hard to write generalized queries for. (There's a hack!)

But I don't remember crashes about null values. 

I feel like sqlite should support nulls cleanly. Else it's impossible to differentiate between an empty string and no data.

Reply:
The crashes were caused by code inside osquery running queries against other tables, then trying to access columns that were not being populated by one or the other implementation.

sqlite does support NULL column values, but osquery was initially designed to always convert everything to string, causing that feature (and real types) to get lost.